src/lib/menu/menu-item.ts

```diff
@@ -26,15 +27,22 @@ export const _MdMenuItemMixinBase = mixinDisabled(MdMenuItemBase);
   host: {
     'role': 'menuitem',
     'class': 'mat-menu-item',
+    '[class.mat-menu-item-highlighted]': '_highlighted',
     '[attr.tabindex]': '_getTabIndex()',
     '[attr.aria-disabled]': 'disabled.toString()',
     '[attr.disabled]': 'disabled || null',
     '(click)': '_checkDisabled($event)',
+    '(mouseenter)': '_emitHoverEvent()',
   },
   templateUrl: 'menu-item.html',
   exportAs: 'mdMenuItem'
 })
-export class MdMenuItem extends _MdMenuItemMixinBase implements Focusable, CanDisable {
+export class MdMenuItem extends _MdMenuItemMixinBase implements Focusable, CanDisable, OnDestroy {
+  /** Stream that emits when the menu item is hovered. */
+  hover: Subject<MdMenuItem> = new Subject();
```

Comment:
It is mainly to facilitate the nested menu, but I can see it being useful in other cases as well.